src/rules/no_empty.rs

```diff
@@ -0,0 +1,314 @@
+// Copyright 2020 the Deno authors. All rights reserved. MIT license.
+use super::{Context, LintRule};
+use swc_ecma_ast::{BlockStmt, Module, SwitchStmt};
+use swc_ecma_visit::{Node, Visit};
+
+pub struct NoEmpty;
+
+impl LintRule for NoEmpty {
+  fn new() -> Box<Self> {
+    Box::new(NoEmpty)
+  }
+
+  fn lint_module(&self, context: Context, module: Module) {
+    let mut visitor = NoEmptyVisitor::new(context);
+    visitor.visit_module(&module, &module);
+  }
+}
+
+pub struct NoEmptyVisitor {
+  context: Context,
+}
+
+impl NoEmptyVisitor {
+  pub fn new(context: Context) -> Self {
+    Self { context }
+  }
+}
+
+impl Visit for NoEmptyVisitor {
+  fn visit_block_stmt(&mut self, block_stmt: &BlockStmt, _parent: &dyn Node) {
+    if block_stmt.stmts.is_empty()
+      && !block_stmt.contains_comments(&self.context)
+    {
+      self.context.add_diagnostic(
+        block_stmt.span,
+        "noEmpty",
+        "Empty block statement",
+      );
+    }
```

Comment:
There should be call to `swc_ecma_visit::visit_block_stmt()` here; otherwise if block is non-empty its contents won't be visited

Reply:
@bartlomieju do you mean to check for nested empty blocks? For example:
```javascript
if (foo) {
  if (bar) {
  }
}
```

Reply:
@DomParfitt yes, I have a suspicion that your example wouldn't be picked up right now

Reply:
Yep, you're right. Just added a test for the above and it currently doesn't report any linting errors. I'll add a check for nested blocks in.